lib/ReadsUtils/ReadsUtilsImpl.py

```diff
@@ -126,32 +126,76 @@ def _proc_upload_reads_params(self, params):
             interleaved = 1
             kbtype = 'KBaseFile.PairedEndLibrary'
             single_end = False
-        seqtype = params.get('sequencing_tech')
-        if not seqtype:
-            raise ValueError('The sequencing technology must be provided')
-
-        sg = 1
-        if 'single_genome' in params and not params['single_genome']:
-            sg = 0
-        o = {'sequencing_tech': seqtype,
-             'single_genome': sg,
-             # 'read_count': params.get('read_count'),
-             # 'read_size': params.get('read_size'),
-             # 'gc_content': params.get('gc_content')
-             }
-        self._add_field(o, params, 'strain')
-        self._add_field(o, params, 'source')
-        ism = params.get('insert_size_mean')
-        self._check_pos(ism, 'insert_size_mean')
-        issd = params.get('insert_size_std_dev')
-        self._check_pos(issd, 'insert_size_std_dev')
-        if not single_end:
-            read_orientation_out = 1 if params.get('read_orientation_outward') else 0
-            o.update({'insert_size_mean': ism,
-                      'insert_size_std_dev': issd,
-                      'interleaved': interleaved,
-                      'read_orientation_outward': read_orientation_out
-                      })
+
+        source_reads_ref = params.get('source_reads_ref')
+        if source_reads_ref:
+            # Means the uploaded reads is a result of an input reads object being filtered/trimmed
+            try:
+                source_reads_object = dfu.get_objects({'object_refs':
+                                                       [source_reads_ref]})['data'][0]
+            except DFUError as e:
+                self.log(('The supplied source_reads_ref {} was not able to be retrieved. ' +
+                          'Logging stacktrace from workspace exception:' +
+                          '\n{}').format(source_reads_ref, e.data))
+                raise
+            # Make sure that no non file related parameters are set. If so throw error.
+            parameters_should_unfilled = ['insert_size_mean', 'insert_size_std_dev',
+                                          'sequencing_tech', 'strain',
+                                          'source', 'read_orientation_outward']
+            if any(x in params for x in parameters_should_unfilled):
+                self.log(("'source_reads_ref' was passed, making the following list of " +
+                          "parameters {} erroneous to " +
+                          "include").format(",".join(parameters_should_unfilled)))
+                raise ValueError(("'source_reads_ref' was passed, making the following list of " +
+                                  "parameters : {} erroneous to " +
+                                  "include").format(", ".join(parameters_should_unfilled)))
```

Comment:
I'd do this check before fetching the object. No need to do the fetch if this check fails, and this check is much faster.